x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/index.tsx

```diff
@@ -465,6 +464,57 @@ export const EndpointList = () => {
     [dispatch]
   );
 
+  const stateToDisplay:
+    | 'loading'
+    | 'policyEmptyState'
+    | 'policyEmptyStateWithoutFleetAccess'
+    | 'hostsEmptyState'
+    | 'endpointTable'
+    | 'listError' = useMemo(() => {
```

Comment:
i'd prefer to leave it as is, because
- we don't need to iterate over the entries for which enum is more useful (but then we could still use `[] as const` to define the values for the union type)
- changing strings in one place but not in another should not be an issue as it throws a type error
- renaming may be a bit problematic, but this is a local variable, so should be easy
- and just learned that union type results in a smaller bundle size that an enum

Reply:
let me know what you think